pkg/resiliency/retry_test.go

```diff
@@ -0,0 +1,80 @@
+package resiliency
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestResiliencyFunctions(t *testing.T) {
+	testCases := []struct {
+		name          string
+		retryOn       []string
+		ignoreOn      []string
+		expectError   bool
+		testCodes     []int32
+		expectedRetry []bool
+	}{
+		{
+			name:          "TestRetryOnHttpCode",
+			retryOn:       []string{"200,201,202,203,204", "300-309", "400", "500,505-509"},
+			expectError:   false,
+			testCodes:     []int32{200, 204, 300, 310, 400, 500, 509},
+			expectedRetry: []bool{true, true, true, false, true, true, true},
+		},
+		{
+			name:          "TestRetryOnGrpcCode",
+			retryOn:       []string{"2", "8-10", "11,13-15"},
+			expectError:   false,
+			testCodes:     []int32{2, 3, 4, 5, 6, 10, 11, 12, 13},
+			expectedRetry: []bool{true, false, false, false, false, true, true, false, true},
+		},
+		{
+			name:          "TestIgnoreOnHttpCode",
+			ignoreOn:      []string{"400-499", "500"},
+			expectError:   false,
+			testCodes:     []int32{200, 300, 400, 500},
+			expectedRetry: []bool{true, true, false, false},
+		},
+		{
+			name:          "TestIgnoreOnGrpcCode",
+			ignoreOn:      []string{"4-10", "15"},
+			expectError:   false,
+			testCodes:     []int32{2, 4, 6, 12, 15},
+			expectedRetry: []bool{true, false, false, true, false},
+		},
+		{
+			name:        "TestRetryOnAndIgnoreOn",
+			retryOn:     []string{"200-299"},
+			ignoreOn:    []string{"200"},
+			expectError: true,
+		},
+		{
+			name:        "TestInvalidStartCodeBiggerThanEndCode",
+			retryOn:     []string{"399-299"},
+			expectError: true,
+		},
+		{
+			name:        "TestInvalidCodeRange",
+			retryOn:     []string{"199-299-300"},
+			expectError: true,
+		},
```

Comment:
Can we add some fuzz tests here https://github.com/google/gofuzz ?

What happens when retry and ignore cover the same range?

Reply:
Sure, I'll add fuzz tests. 
When retry and ignore are used together, an error will occur.

Reply:
@JoshVanL I may not have enough time to complete this test recently, since you mentioned you have a WIP local branch, can you help with this test as well as the integration test?